clients/java/client/src/it/java/org/operaton/bpm/client/variable/FileSerializationIT.java

```diff
@@ -567,32 +569,29 @@ public void shouldFailWhenCompletingWihtDeferredFileValue() {
     ExternalTask fooTask = invocation.getExternalTask();
     ExternalTaskService fooService = invocation.getExternalTaskService();
 
-
-    // then
-    assertThatThrownBy(() ->
-            client.subscribe(EXTERNAL_TASK_TOPIC_BAR)
-                    .handler(handler)
-                    .open()).isInstanceOf(ValueMapperException.class);
+    client.subscribe(EXTERNAL_TASK_TOPIC_BAR)
+      .handler(handler)
+      .open();
 
     // when
     Map<String, Object> variables = new HashMap<>();
     DeferredFileValue deferredFileValue = fooTask.getVariableTyped(VARIABLE_NAME_FILE);
     variables.put("deferredFile", deferredFileValue);
     variables.put(ANOTHER_VARIABLE_NAME_FILE, ANOTHER_VARIABLE_VALUE_FILE);
-    fooService.complete(fooTask, variables);
 
-    clientRule.waitForFetchAndLockUntil(() -> !handler.getHandledTasks().isEmpty());
+    // then
+    assertThatThrownBy(() -> fooService.complete(fooTask, variables))
+            .isInstanceOf(ValueMapperException.class);
 
     // then
     List<VariableInstanceDto> variableInstances = engineRule.getVariablesByProcessInstanceIdAndVariableName(processInstance.getId(), null);
-    assertThat(variableInstances).hasSize(2);
 
     List<String> variableNames = new ArrayList<>();
     for (VariableInstanceDto variableInstance : variableInstances) {
       variableNames.add(variableInstance.getName());
     }
 
-    assertThat(variableNames).containsExactlyInAnyOrder(VARIABLE_NAME_FILE, ANOTHER_VARIABLE_NAME_FILE); // contains not "deferredFile"
```

Comment:
This nearly drove me mad. I'm suspecting, that due to the expected thrown error in the previous JUnit 4 tests, there was an error introduced in the test.

If we complete an external task service with two variables, of which one causes a serialization error, my expectation would be that the transaction is rolled back in the db and NONE of the two variables is in the database. I cannot understand the usecase which makes one of two variables and up in the database.

@a-m-zill can I get an opinion here?

Reply:
Hi @javahippie,
as we have talked about in our chat, i agree that this is not a right behavior to expect one variable saved and one not of two new variables in one transaction.
Further i suggest to check in the end of the test that the process instance is again or still on the external task. Because that it the position where it should be.

Reply:
Hi, the documentation for the ExpectedException rule says: 
"This rule does not perform any special magic to make execution continue as if the exception had not been thrown. So it is nearly always a mistake for a test method to have statements after the one that is expected to throw the exception"
So it is very likely that variables were never saved and the subsequent test assertion never happened, but the test still reported success because the expected exception was thrown as intended. So I think your expectations are correct and the test had unreachable false assertions

Reply:
@shiftyseaotter thanks for reading the docs and finding this! This might have implications on other tests which use / used `ExcpectedException`. @kthoms Something to keep in mind for upcoming refactorings

Reply:
@a-m-zill Added a check for the currently active topic, and refactored the code a little, happy about an approve if it works for you!